components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/GatewayArtifactsMgtDAO.java

```diff
@@ -517,6 +517,75 @@ public List<APIRuntimeArtifactDto> retrieveGatewayArtifactsByAPIIDs(List<String>
         return apiRuntimeArtifactDtoList;
     }
 
+    public List<APIRuntimeArtifactDto> retrieveGatewayArtifactsOnlyByAPIIDs(List<String> apiIds, String tenantDomain)
+            throws APIManagementException {
+        // Logging the API ID List
+        log.info("Getting runtime artifacts for the API ID List: " + apiIds.toString());
+        // Split apiId list into smaller list of size 25
+        List<List<String>> apiIdsChunk = new ArrayList<>();
+        int apiIdListSize = apiIds.size();
+        int apiIdArrayIndex = 0;
+        int apiIdsChunkSize = SQLConstants.API_ID_CHUNK_SIZE;
+        while (apiIdArrayIndex < apiIdListSize) {
+            apiIdsChunk.add(apiIds.subList(apiIdArrayIndex, Math.min(apiIdArrayIndex + apiIdsChunkSize, apiIdListSize)));
+            apiIdArrayIndex += apiIdsChunkSize;
+        }
```

Comment:
This can also be done by using streams. See below example:
https://stackoverflow.com/a/69635781/3786453

Reply:
Updated: https://github.com/wso2/carbon-apimgt/pull/11304/commits/c1dd6055755e9d451636e15a814c7aff415a99e1